src/software/estop/UARTCommunication.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <boost/asio.hpp>
+#include <boost/function.hpp>
+
+/*
+ * A boost::asio wrapper class that is used to synchronously communicate with a serial device via UART
+ */
+typedef boost::asio::io_service io_service;
+typedef std::shared_ptr<boost::asio::serial_port> serial_port_ptr;
```

Comment:
since these are only used in the `UARTCommunication` class, we should put these into the class itself so that they don't appear in the global scope. 

nit: it's better to do `using serial_port_ptr = std::shared_ptr<...>`  instead of `typedef`.

Reply:
changed both `typedef` to `using` and placed inside the class